### DISCUSSION TITLE
node_modules compile error

### DISCUSSION THREAD

**Discussion by Burrito08** [citizenfx/fivem#2170] [Engineering Discussion]
Hi I get the error when compiling in ts 
node_modules/@citizenfx/server/index.d.ts:133:5 - error TS2374: Duplicate index signature for type 'string'.

**Comment by nihonium-cfx** [comment]
Hi there! May I ask you what versions of TypeScript and this package are you using that gives this error?

**Comment by Burrito08** [comment]
Hi typescript 5.1.6 my tsconfig {
  "compilerOptions": {
    "types": ["@citizenfx/client", "@citizenfx/server"],
    "removeComments": true,
    "noUnusedLocals": false,
    "noUnusedParameters": false,
    "noImplicitReturns": true,
    "moduleResolution": "Node",
    "resolveJsonModule": true,
    "allowJs": true,
    "emitDecoratorMetadata": false,
    "experimentalDecorators": true,
    "target": "ES2017",                                
    "module": "commonjs",                               
    "rootDir": "./src",  
    "outDir": "./resources",                              
    "allowSyntheticDefaultImports": true,                               
  },
  "include": ["./src/**/*.ts"],
  "exclude": ["node_modules"]
}

**Comment by nihonium-cfx** [comment]
All right, the error you're getting is due to that you've included both client and server typings simultaneously, but they're not meant to be used like that exactly because they define the very same common types and that leads to such an error.

What you can do here is either:
- Separate TS projects (tsconfig) for client and server, that'd be the best way right now.
- Include appropriate typings where you need them, so for example if X script is meant to be run on client - reference `@citizenfx/client` there.

**Comment by Burrito08** [comment]
thank you
